src/libraries/view-timeseries-graph/TimeseriesGraphView.tsx

```diff
@@ -0,0 +1,210 @@
+import { TimeseriesGraphViewData } from './TimeseriesGraphViewData'
+
+type Props = {
+    data: TimeseriesGraphViewData
+    timeseriesLayoutOpts?: TimeseriesLayoutOpts
+    width: number
+    height: number
+}
+
+type PanelProps = {
+    pixelZero: number
+    dimensions: {
+        dimensionIndex: number
+        dimensionLabel: string
+        pixelTimes: number[]
+        pixelValues: number[]
+        type: string
+        attributes: {[key: string]: any}
+    }[]
+}
+
+const panelSpacing = 4
+const emptyPanelSelection = new Set<number | string>()
+
+const TimeseriesGraphView: FunctionComponent<Props> = ({data, timeseriesLayoutOpts, width, height}) => {
+    const {datasets, series} = data
+
+    const resolvedSeries = useMemo(() => (
+        series.map(s => {
+            const ds = datasets.filter(d => (d.name === s.dataset))[0]
+            if (ds === undefined) throw Error(`Dataset not found in series: ${s.dataset}`)
+            return {
+                ...s,
+                t: ds.data[s.encoding['t']],
+                y: ds.data[s.encoding['y']]
+            }
+        })
+    ), [series, datasets])
+
+    const {minTime, maxTime} = useMemo(() => (
+        {
+            minTime: min(resolvedSeries.map(s => (min(s.t)))),
+            maxTime: max(resolvedSeries.map(s => (max(s.t))))
+        }
+    ), [resolvedSeries])
+
+    const {minValue, maxValue} = useMemo(() => (
+        {
+            minValue: min(resolvedSeries.map(s => (min(s.y)))),
+            maxValue: max(resolvedSeries.map(s => (max(s.y))))
+        }
+    ), [resolvedSeries])
+
+    // This component ignores timeOffset except in the following two hooks
+    useRecordingSelectionTimeInitialization(minTime, maxTime, 0)
+    const {visibleTimeStartSeconds, visibleTimeEndSeconds } = useTimeRange(0) // timeOffset is subtracted from start and end after getting from the global state
+
+    const margins = useTimeseriesMargins(timeseriesLayoutOpts)
+
+    // Compute the per-panel pixel drawing area dimensions.
+    const panelCount = 1
+    const toolbarWidth = timeseriesLayoutOpts?.hideToolbar ? 0 : DefaultToolbarWidth
+    const { panelWidth, panelHeight } = usePanelDimensions(width - toolbarWidth, height, panelCount, panelSpacing, margins)
+
+    // We need to have the panelHeight before we can use it in the paint function.
+    // By using a callback, we avoid having to complicate the props passed to the painting function; it doesn't make a big difference
+    // but simplifies the prop list a bit.
+    const paintPanel = useCallback((context: CanvasRenderingContext2D, props: PanelProps) => {
+        // don't display dashed zero line (Eric's request)
+        // context.strokeStyle = 'black'
+        // context.setLineDash([5, 15]);
+        // context.lineWidth = 1
+        // context.beginPath()
+        // context.moveTo(0, props.pixelZero)
+        // context.lineTo(panelWidth, props.pixelZero)
+        // context.stroke()
+        // context.setLineDash([]);
```

Comment:
Might be worth making this a toggle? (Whether at UI or code level)

Reply:
I think we should wait until we have a need for that.